src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -700,6 +701,289 @@ describe('Transactions Controller (Unit)', () => {
     });
   });
 
+  describe('Preview transactions', () => {
+    it('should throw a validation error', async () => {
+      const previewTransactionDto = previewTransactionDtoBuilder().build();
+      await request(app.getHttpServer())
+        .post(
+          `/v1/chains/${faker.random.numeric()}/transactions/${faker.datatype.hexadecimal(
+            16,
+          )}/preview`,
+        )
+        .send({ ...previewTransactionDto, value: 1 })
+        .expect(400);
+    });
+
+    it('should preview a transaction', async () => {
+      const previewTransactionDto = previewTransactionDtoBuilder().build();
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const safeResponse = safeBuilder().with('address', safeAddress).build();
+      const chainResponse = chainBuilder().build();
+      const dataDecodedResponse = dataDecodedBuilder().build();
+      const contractResponse = contractBuilder().build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrl = `${safeConfigApiUrl}/api/v1/chains/${chainId}`;
+        const getSafeUrl = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}`;
+        const getContractUrlPattern = `${chainResponse.transactionService}/api/v1/contracts/`;
+        if (url === getChainUrl) {
+          return Promise.resolve({ data: chainResponse });
+        }
+        if (url === getSafeUrl) {
+          return Promise.resolve({ data: safeResponse });
+        }
+        if (url.includes(getContractUrlPattern)) {
+          return Promise.resolve({ data: contractResponse });
+        }
+        return Promise.reject(new Error(`Could not match ${url}`));
+      });
+      mockNetworkService.post.mockImplementation((url) => {
+        const getDataDecodedUrl = `${chainResponse.transactionService}/api/v1/data-decoder/`;
+        if (url === getDataDecodedUrl) {
+          return Promise.resolve({ data: dataDecodedResponse });
+        }
+        return Promise.reject(new Error(`Could not match ${url}`));
+      });
+      await request(app.getHttpServer())
+        .post(`/v1/chains/${chainId}/transactions/${safeAddress}/preview`)
+        .send(previewTransactionDto)
+        .expect(200)
+        .expect(({ body }) => {
+          expect(body).toMatchObject({
```

Comment:
Why do we use `toMatchObject` here? It seems that it's missing just two properties but is it because of that?

Reply:
> Yes, it was used to abstract from those two property values in the tests which are not strictly related to them. But I've changed the tests to have the complete expected payload (setting more values into `previewTransactionDtoBuilder` call) so the tests are a bit more strict: [a158c2a](https://github.com/5afe/safe-client-gateway-nest/pull/291/commits/a158c2a02ceed6346e496ef948aac0185e20225d)

I think the way it was before was also ok! Just clarifying if that was the case. I think we should use `toMatchObject` if we already test the complete payload in some other test (this reduces the size of the test even though not by a lot in this case) 👍